Core/src/org/sleuthkit/autopsy/centralrepository/contentviewer/DataContentViewerOtherCases.java

```diff
@@ -414,64 +422,130 @@ private AbstractFile getAbstractFileFromNode(Node node) {
         } catch (NoCurrentCaseException ex) {
             LOGGER.log(Level.SEVERE, "Exception while getting open case.", ex); // NON-NLS
+        } catch (TskCoreException ex) {
+            // do nothing. 
+            // @@@ Review this behavior
         }
 
-        return Collections.emptyList();
+        return new HashMap<>(0);
     }
 
-    @Override
-    public boolean isSupported(Node node) {
-        if (!EamDb.isEnabled()) {
-            return false;
+    private List<AbstractFile> addCaseDbMatches(CorrelationAttribute corAttr, Case openCase) throws NoCurrentCaseException, TskCoreException, EamDbException {
+        String md5 = corAttr.getCorrelationValue();
+
+        SleuthkitCase tsk = openCase.getSleuthkitCase();
+        List<AbstractFile> matches = tsk.findAllFilesWhere(String.format("md5 = '%s'", new Object[]{md5}));
+
+        List<AbstractFile> caseDbArtifactInstances = new ArrayList<>();
+        for (AbstractFile fileMatch : matches) {
+            if (this.file.equals(fileMatch)) {
+                continue; // If this is the file the user clicked on
+            }
+            caseDbArtifactInstances.add(fileMatch);
+        }
+        return caseDbArtifactInstances;
+
+    }
+
+    private void addOrUpdateAttributeInstance(final Case openCase, Map<ArtifactKey, CorrelationAttributeInstance> artifactInstances, AbstractFile caseDbFile) throws TskCoreException, EamDbException {
+        CorrelationCase caze = new CorrelationCase(openCase.getNumber(), openCase.getDisplayName());
+        CorrelationDataSource dataSource = CorrelationDataSource.fromTSKDataSource(caze, caseDbFile.getDataSource());
+        String filePath = caseDbFile.getParentPath() + caseDbFile.getName();
+        ArtifactKey instKey = new ArtifactKey(dataSource.getDeviceID(), filePath);
+        CorrelationAttributeInstance caseDbInstance = new CorrelationAttributeInstance(caze, dataSource, filePath, "", caseDbFile.getKnown());
+        TskData.FileKnown knownStatus = caseDbInstance.getKnownStatus();
+        // If not known, check Tags for known and set
+        if (knownStatus.compareTo(TskData.FileKnown.KNOWN) != 0 && knownStatus.compareTo(TskData.FileKnown.BAD) != 0) {
+            List<ContentTag> fileMatchTags = openCase.getServices().getTagsManager().getContentTagsByContent(caseDbFile);
+            for (ContentTag tag : fileMatchTags) {
+                TskData.FileKnown tagKnownStatus = tag.getName().getKnownStatus();
+                if (tagKnownStatus.compareTo(TskData.FileKnown.KNOWN) == 0 || tagKnownStatus.compareTo(TskData.FileKnown.BAD) == 0) {
+                    caseDbInstance.setKnownStatus(TskData.FileKnown.BAD);
+                    break;
+                }
+            }
+        }
+
+        // If known, or not in CR, add
+        if (knownStatus.compareTo(TskData.FileKnown.KNOWN) == 0 || knownStatus.compareTo(TskData.FileKnown.BAD) == 0 || !artifactInstances.containsKey(instKey)) {
```

Comment:
I feel like using "equals" instead of "compareTo" would make this code (and other similar code) more readable. This would change to: 
if (knownStatus.equals(TskData.FileKnown.KNOWN) || knownStatus.equals(TskData.FileKnown.BAD) || ...